components/script/dom/node.rs

```diff
@@ -118,6 +124,9 @@ pub struct Node {
     /// The document that this node belongs to.
     owner_doc: MutNullableDom<Document>,
 
+    /// Rare node data.
+    rare_data: Box<NodeRareData>,
```

Comment:
Oh, I totally missed this, the point of this is that this should be an `Option<Box<>>` (or such), and be `None` when not needed. otherwise this doesn't help reduce memory usage at all, mostly the opposite.